helpers/hcache_simple.py

```diff
@@ -491,19 +491,43 @@ def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
             func_name = func_name[: -len("_intrinsic")]
         set_cache_property("system", func_name, "type", cache_type)
 
+
         @functools.wraps(func)
-        def wrapper(*args: Any, **kwargs: Any) -> Any:
+        def wrapper(
+            force_refresh= False,
+            abort_on_cache_miss = False,
+            report_on_cache_miss= False,
+            *args: Any,
+            **kwargs: Any,) -> Any:
             # Get the function name.
             func_name = func.__name__
             if func_name.endswith("_intrinsic"):
                 func_name = func_name[: -len("_intrinsic")]
             # Get the cache.
             cache = get_cache(func_name)
+
+            if func_name == "_call_api_sync_cached":
```

Comment:
This seems custom? I'm ok if there is a general case and we can decorate the function with a special thing.

A better alternative is to use an option through the interface, instead of having the code infers stuff from the name.

`_intrinsic` is kind of ok since it's a general approach, but this one calls for a switch.